lib/Dialect/FIRRTL/Transforms/LowerAnnotations.cpp

```diff
@@ -473,12 +476,33 @@ LogicalResult LowerAnnotationsPass::applyAnnotation(DictionaryAttr anno,
   return success();
 }
 
+/// Convert consumed SourceAnnotation and SinkAnnotation into WiringProblems,
+/// using the pin attribute as newNameHint
+LogicalResult LowerAnnotationsPass::legacyToWiringProblems(ApplyState &state) {
+  for (const auto &problem : state.legacyWiringProblems) {
```

Comment:
Consider structured binding here, if you like it better:
```
for (const auto &[name, problem] : state.legacyWiringProblems) {
```
Don't usually use ref for Attribute's but doesn't hurt and no need to copy the problem.